src/contexts/Global/index.tsx

```diff
@@ -0,0 +1,294 @@
+import * as React from 'react'
+import PropTypes from 'prop-types'
+import Router from 'next/router'
+import { useRemoteConfig } from '../RemoteConfig'
+
+interface GlobalState {
+  /**
+   * A `boolean` which indicates whether the cart menu is open or not.
+   * The cart menu is the menu that appears when the user clicks on the cart icon.
+   * @default false
+   */
+  isCartMenuOpen: boolean
+  /**
+   * A `boolean` which indicates whether the cookie bar is open or not.
+   * The cookie bar is the bar that appears when the user enters the website
+   * for the first time.
+   *
+   * @note This value is set to `true` after a delay of `COOKIE_BAR_ENTER_DELAY`,
+   * provided the user has not previously dismissed the cookie bar.
+   * @default false
+   */
+  isCookieBarOpen: boolean
+  /**
+   * A `boolean` which indicates whether the market menu is open or not.
+   * The market menu contains the list of markets that the user can switch between.
+   * @default false
+   */
+  isMarketMenuOpen: boolean
+  /**
+   * A `boolean` which indicates whether the navigation menu is open or not.
+   * The navigation menu is the menu that appears when the user clicks on the
+   * three stacked lines icon, or otherwise known as the _hamburger_ icon.
+   * @default false
+   */
+  isNavMenuOpen: boolean
+  /**
+   * A `boolean` which indicates whether the search menu is open or not.
+   * The search menu is the menu that appears when the user clicks on the search
+   * icon.
+   * @default false
+   */
+  isSearchMenuOpen: boolean
+  /**
+   * A `boolean` which indicates whether the store message is open or not.
+   * The store message is usually located above the site header and is used to
+   * display important messages to the user.
+   * @default false
+   */
+  isStoreMessageOpen: boolean
+  /**
+   * A `boolean` which indicates whether any of the following menu's are open:
+   * - `isCartMenuOpen`
+   * - `isNavMenuOpen`
+   * - `isSearchMenuOpen`
+   * @default false
+   */
+  isSomeMenuOpen: boolean
+}
+
+type GlobalStateCTX = GlobalState | undefined
+export const GlobalStateContext = React.createContext<GlobalStateCTX>(undefined)
+
+interface GlobalHandlers {
+  /**
+   * A setter function which sets the `isCartMenuOpen` state to `false`.
+   */
+  onCartMenuClose: () => void
+  /**
+   * A setter function which toggles the `isCartMenuOpen` state.
+   * @note This function also sets the `isNavMenuOpen` and `isSearchMenuOpen`
+   * states to `false`.
+   */
```

Comment:
> @note This function also sets the `isNavMenuOpen` and `isSearchMenuOpen`
> states to `false`.

Wonder if this is too prone for invalidation as well? 🤔

Reply:
I am not a fan of this. As you say, it's too error-prone and may incite false confidence on the implementor that things will behave a certain way. I'd rather be more imperative here and have each setter do one thing.

Reply:
Do we want to address this already in this PR or leave that for a future PR?